spec/sidekiq-instrument/worker_spec.rb

```diff
@@ -117,4 +119,101 @@
       ]
     end
   end
+
+  describe 'client & server middleware' do
+    before(:each) do
+      Redis.new.flushall
+      Sidekiq.configure_client do |c|
+        c.client_middleware do |chain|
+          chain.add Sidekiq::Instrument::ClientMiddleware
+        end
+      end
+    end
+
+    after(:each) do
+      Sidekiq.configure_client do |c|
+        c.client_middleware do |chain|
+          chain.remove Sidekiq::Instrument::ClientMiddleware
+        end
+      end
+    end
+
+    context 'successful increment' do
+      let(:expected_dog_options) { { tags: ['queue:default', 'worker:my_worker'] } }
+      let(:worker_metric_name) do
+        "sidekiq_instrument_trace_workers::in_queue"
+      end
+
+      before do
+        Sidekiq.server_middleware do |chain|
+          chain.add Sidekiq::Instrument::ServerMiddleware
+        end
+      end
+  
+      after do
+        Sidekiq.server_middleware do |chain|
+          chain.remove Sidekiq::Instrument::ServerMiddleware
+        end
+      end
+
+      it 'increments the in queue counter' do
+          Sidekiq::Instrument::WorkerMetrics.enabled = true
+          redis = Redis.new
+          expect(redis.hget(worker_metric_name ,'my_worker')).to be nil
+          MyWorker.perform_async
+          expect(redis.hget(worker_metric_name ,'my_worker')).to eq('1')
+          MyWorker.perform_async
+          expect(redis.hget(worker_metric_name ,'my_worker')).to eq('2')
```

Comment:
Can we "de-indent" this block by one (there are quite a few blocks like this; it'd be nice to sweep for them all, even if they aren't part of the changes here that I can remark upon)?
```suggestion
        Sidekiq::Instrument::WorkerMetrics.enabled = true
        redis = Redis.new
        expect(redis.hget(worker_metric_name ,'my_worker')).to be nil
        MyWorker.perform_async
        expect(redis.hget(worker_metric_name ,'my_worker')).to eq('1')
        MyWorker.perform_async
        expect(redis.hget(worker_metric_name ,'my_worker')).to eq('2')
```